### PR TITLE
[FIX](complex-type)fix complex type nested col_const

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx
for array/map/struct in mysql_writer unpack_if_const only unpack self column not nested , so col_const should not used in nested column.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.92 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17162305740 Bytes

**Comment by xiaokang** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.89 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         30 seconds loaded 2358488459 Bytes, about 74 MB/s
 stream load orc:          79 seconds loaded 1101869774 Bytes, about 13 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17162606505 Bytes

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by xiaokang** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.7 seconds
 stream load tsv:          503 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          36.2 seconds inserted 10000000 Rows, about 276K ops/s
 storage size: 17161751762 Bytes

File: regression-test/data/query_p0/sql_functions/array_functions/test_array_with_scale_type.out [hunk 1]
```diff
@@ -58,8 +58,8 @@
 [23.679]
 
 -- !select --
-[24.990, 24.990]
-[24.990, 24.990]
+[24.990, 25.990]
```
Comment (yiguolei) [inline regression-test/data/query_p0/sql_functions/array_functions/test_array_with_scale_type.out:61]: why these outputs also changed?

File: regression-test/suites/query_p0/sql_functions/struct_functions/test_struct_functions_by_literal.groovy [hunk 2]
```diff
@@ -16,6 +16,9 @@
 // under the License.
 
 suite("test_struct_functions_by_literal") {
+    // struct-nested
+    qt_sql "select s from (select struct('a', 1, 'doris', 'aaaaa', 1.32) as s) t"
```
Comment (yiguolei) [inline regression-test/suites/query_p0/sql_functions/struct_functions/test_struct_functions_by_literal.groovy:20]: what's the meaning of this struct?

File: regression-test/data/query_p0/sql_functions/array_functions/test_array_with_scale_type.out [hunk 3]
```diff
@@ -163,8 +163,8 @@
 23.679	[23.679, 34.679]	[23.679, 23.679, 34.679]
 
 -- !select --
-[2022-12-02 22:23:24.999, 2022-12-02 22:23:24.999]
-[2022-12-02 22:23:24.999, 2022-12-02 22:23:24.999]
+[2022-12-02 22:23:24.999, 2022-12-02 22:23:23.997]
```
Comment (yiguolei) [inline regression-test/data/query_p0/sql_functions/array_functions/test_array_with_scale_type.out:166]: why thest output changed?

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**